scripts/west_commands/runners/uf2.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) 2023 Peter Johanson <peter@peterjohanson.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''UF2 runner (flash only) for UF2 compatible bootloaders.'''
+
+from pathlib import Path
+from shutil import copy
+
+from runners.core import ZephyrBinaryRunner, RunnerCaps
+
+try:
+    import psutil  # pylint: disable=unused-import
+    MISSING_PSUTIL = False
+except ImportError:
+    # This can happen when building the documentation for the
+    # runners package if edtlib is not on sys.path. This is fine
+    # to ignore in that case.
+    MISSING_PSUTIL = True
+
+class UF2BinaryRunner(ZephyrBinaryRunner):
+    '''Runner front-end for bossac.'''
+
+    def __init__(self, cfg, board_id=None):
+        super().__init__(cfg)
+        self.board_id = board_id
+
+    @classmethod
+    def name(cls):
+        return 'uf2'
+
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={'flash'})
+
+    @classmethod
+    def do_add_parser(cls, parser):
+        parser.add_argument('--board-id', required=False, dest='board_id',
+                            help='Board-ID value to match from INFO_UF2.TXT')
+
+    @classmethod
+    def do_create(cls, cfg, args):
+        return UF2BinaryRunner(cfg, board_id=args.board_id)
+
+    def is_uf2_partition(self, part):
+        try:
+            return (Path(part.mountpoint) / "INFO_UF2.TXT").is_file()
+        except PermissionError:
+            return False
+
+    def get_uf2_info(self, part):
+        lines = (Path(part.mountpoint) / "INFO_UF2.TXT").read_text().splitlines()
+
+        lines.pop(0) # Pop off the summary line
+
+        return dict((p[0], p[1]) for p in [p.strip().split(': ', 2) for p in lines])
```

Comment:
This would fail if a line doesn't contain a `: `. 

Alternate implementation idea:

```python
return {k.strip(): v.strip() for k, _, v in (line.partition(':') for line in lines)}
```

Version that omits lines with no key/value:
```python
def split_uf2_info(line: str):
    k, _, v = line.partition(':')
    return k.strip(), v.strip()

return {k: v for k, v in (split_uf2_info(line) for line in lines) if k and v}
```

Reply:
Used the second version to be safe. Thanks!